### DISCUSSION TITLE
Breakout: maintenance of unofficial extension points

### DISCUSSION THREAD

**Discussion by teunbrand** [ggplot2-extenders/ggplot-extension-club#69] [General]
One of the things I find lackluster is that github discussion appear highly linear, with no space to manouver into tangents without affecting the main discussion. So this is an attempt to create space for a tangent.

Coined by June here: https://github.com/teunbrand/ggplot-extension-club/discussions/68#discussioncomment-11573074
> "Assuming you take the route of deviating from the advertised extension mechanism and extending an unofficial extension point, what is the most principled way to maintain this?"

**Comment by yjunechoe** [comment]
You know, soon after I commented that I thought "should I have just started a new thread?" but now I'm outside and didn't want to risk fiddling with a delete and copy paste on mobile, so thanks for this 😅

**Comment by teunbrand** [comment]
I don't have a strictly curated opinion on this as a whole yet, so I'll just open with an example of what I think was a good thing to do.

So there was this one time I was investigating ggplot2 revdep failures (~~I think~~ it was in ggdag (I've checked now)), and they extended the ScalesList object. The extension was basically a STFU extension, overriding the `ScalesList$add()` method to silence warnings. So included in the PR was a small snapshot test that I called a 'sentinel test'.

https://github.com/r-causal/ggdag/blob/46d7ca800ae1f617a4a2f8c33f68dc7f644b59c6/tests/testthat/test-StatsandGeoms.R#L18-L25.

It is basically a 'canary in the coalmine' so you know when that snapshot fails, there have been upstream changes. That of course leaves the possiblity for a false positive failure, but at least you have the finger on the pulse in some sense.

**Comment by teunbrand** [comment]
Another example perhaps is the `scale_x/y_facet()` in ggh4x. These use `ggplot_add()` as a Trojan horse to 'infect' the `Facet*` of a plot and override some of the facet methods to use the scales. Of course there might be extensions that override facets, so I do a little check to see if the methods I want to override are indeed the vanilla methods and thus my assumptions hold.

https://github.com/teunbrand/ggh4x/blob/af4e3dfabd2f2203da22e86c5ef35dfbd13d47a5/R/facetted_pos_scales.R#L205-L218

This is more of an on-the-fly test than a proper unit-test though.
